core/consensus/babe/src/lib.rs

```diff
@@ -929,29 +930,13 @@ impl MaybeSpanEpoch {
 	}
 }
 
-/// Extract current epoch data from cache and fallback to querying the runtime
-/// if the cache isn't populated.
-fn epoch<B, C>(client: &C, at: &BlockId<B>) -> Result<MaybeSpanEpoch, ConsensusError> where
+/// Extract current epoch data from the runtime.
+fn epoch<B, C>(client: &C, at: &BlockId<B>) -> Option<MaybeSpanEpoch> where
 	B: BlockT,
 	C: ProvideRuntimeApi + ProvideCache<B>,
 	C::Api: BabeApi<B>,
 {
-	epoch_from_cache(client, at)
-		.or_else(|| epoch_from_runtime(client, at).map(MaybeSpanEpoch::Regular))
-		.ok_or(consensus_common::Error::InvalidAuthoritiesSet)
-}
-
-/// Extract current epoch data from cache.
-fn epoch_from_cache<B, C>(client: &C, at: &BlockId<B>) -> Option<MaybeSpanEpoch> where
-	B: BlockT,
-	C: ProvideCache<B>,
-{
-	// the epoch that is BABE-valid at the block is not the epoch that is cache-valid at the block
-	// we need to go back for maximum two steps
-	client.cache()
-		.and_then(|cache| cache
-			.get_at(&well_known_cache_keys::EPOCH, at)
-			.and_then(|(_, _, v)| Decode::decode(&mut &v[..]).ok()))
+	epoch_from_runtime(client, at).map(MaybeSpanEpoch::Regular)
```

Comment:
This will break light clients.  If this is intentional, that is fine, but it must be documented, both in the rustdoc and in the `--help` output and/or man pages of the final binary.  Additionally, I believe that there should be an issue filed for light client support.

If light client support was already broken, then this should be documented too.

Reply:
Light clients are tagged as `Experimental` - they aren't guaranteed to work or to continue to work.